ports/stm32/boards/ARDUINO_OPTA/mpconfigboard.h

```diff
@@ -0,0 +1,239 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ * The MIT License (MIT)
+ * Copyright (c) 2024 Arduino SA
+ */
+
+#define MICROPY_HW_BOARD_NAME       "Arduino OPTA"
+#define MICROPY_HW_MCU_NAME         "STM32H747"
+#define MICROPY_HW_FLASH_FS_LABEL   "OPTA"
+
+// Network config
+#define MICROPY_PY_NETWORK_HOSTNAME_DEFAULT "mpy-opta"
+
+#define MICROPY_OBJ_REPR            (MICROPY_OBJ_REPR_C)
+#define UINT_FMT                    "%u"
+#define INT_FMT                     "%d"
+typedef int mp_int_t;               // must be pointer size
+typedef unsigned int mp_uint_t;     // must be pointer size
+
+#define MICROPY_FATFS_EXFAT         (1)
+#define MICROPY_HW_ENABLE_RTC       (1)
+#define MICROPY_HW_ENABLE_RNG       (1)
+#define MICROPY_HW_ENABLE_ADC       (1)
+#define MICROPY_HW_ENABLE_DAC       (0)
+#define MICROPY_HW_ENABLE_USB       (1)
+#define MICROPY_HW_HAS_SWITCH       (1)
+#define MICROPY_HW_HAS_FLASH        (1)
+#define MICROPY_HW_ENABLE_SERVO     (0)
+#define MICROPY_HW_ENABLE_TIMER     (1)
+#define MICROPY_HW_ENABLE_SDCARD    (0)
+#define MICROPY_HW_ENABLE_MMCARD    (0)
+
+// Flash storage config
+#define MICROPY_HW_SPIFLASH_ENABLE_CACHE            (1)
+#define MICROPY_HW_ENABLE_INTERNAL_FLASH_STORAGE    (0)
+
+#define MICROPY_BOARD_STARTUP       OPTA_board_startup
+void OPTA_board_startup(void);
+
+#define MICROPY_BOARD_EARLY_INIT    OPTA_board_early_init
+void OPTA_board_early_init(void);
+
+#define MICROPY_HW_ENTER_BOOTLOADER_VIA_RESET   (0)
+#define MICROPY_BOARD_ENTER_BOOTLOADER(nargs, args) OPTA_board_enter_bootloader()
+void OPTA_board_enter_bootloader(void);
+
+void OPTA_board_low_power(int mode);
+#define MICROPY_BOARD_LEAVE_STOP    OPTA_board_low_power(0);
+#define MICROPY_BOARD_ENTER_STOP    OPTA_board_low_power(1);
+#define MICROPY_BOARD_ENTER_STANDBY OPTA_board_low_power(2);
+
+void OPTA_board_osc_enable(int enable);
+#define MICROPY_BOARD_OSC_ENABLE    OPTA_board_osc_enable(1);
+#define MICROPY_BOARD_OSC_DISABLE   OPTA_board_osc_enable(0);
```

Comment:
I don't think these two macros are used anywhere?

Reply:
They're not currently used upstream but I use them in stop mode. I can send this patch too if you like, it will not have any effect on other boards:

```diff
diff --git a/ports/stm32/powerctrl.c b/ports/stm32/powerctrl.c
index 775514922..38e44faaf 100644
--- a/ports/stm32/powerctrl.c
+++ b/ports/stm32/powerctrl.c
@@ -823,6 +823,10 @@ void powerctrl_enter_stop_mode(void) {
     }
     #endif
 
+    #if defined(MICROPY_BOARD_OSC_DISABLE)
+    MICROPY_BOARD_OSC_DISABLE
+    #endif
+
     #if defined(STM32WB)
     powerctrl_low_power_prep_wb55();
     #endif
@@ -833,6 +837,10 @@ void powerctrl_enter_stop_mode(void) {
     HAL_PWR_EnterSTOPMode(PWR_LOWPOWERREGULATOR_ON, PWR_STOPENTRY_WFI);
     #endif
 
+    #if defined(MICROPY_BOARD_OSC_ENABLE)
+    MICROPY_BOARD_OSC_ENABLE
+    #endif
+
     // reconfigure the system clock after waking up
 
     #if defined(STM32F0)
```

Reply:
That patch is useful.  But I'd suggest naming the macros something like `MICROPY_BOARD_PRE_STOP` and `MICROPY_BOARD_POST_STOP`.  That allows also having a `MICROPY_BOARD_PRE_STANDBY`.